From e6acba323a06781f4751a05b1bb0b8cbe8834636 Mon Sep 17 00:00:00 2001
From: Wen He <wen.he_1@nxp.com>
Date: Mon, 12 Aug 2019 10:18:20 +0800
Subject: [PATCH 581/741] arm64: dts: ls1028a: Add properties node for Display
 output pixel clock

commit 8c10599d5fc0700ebeeb895c551e462a1a49aa7f from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The LS1028A has a clock domain PXLCLK0 used for the Display output
interface in the display core, independent of the system bus frequency,
for flexible clock design. This display core has its own pixel clock.

This patch enable the pixel clock provider on the LS1028A.

Signed-off-by: Wen He <wen.he_1@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 31c3449cc91d..117126e523a1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -76,11 +76,18 @@
 		clock-output-names = "sysclk";
 	};
  
-        dpclk: clock-dp {
+        osc_27m: clock-osc-27m {
                 compatible = "fixed-clock";
                 #clock-cells = <0>;
                 clock-frequency = <27000000>;
-                clock-output-names= "dpclk";
+		clock-output-names = "phy_27m";
+        };
+
+        dpclk: clock-display@f1f0000 {
+                compatible = "fsl,ls1028a-plldig";
+                reg = <0x0 0xf1f0000 0x0 0xffff>;
+                #clock-cells = <0>;
+                clocks = <&osc_27m>;
         };
 
         aclk: clock-axi {
-- 
2.17.1

